test/unit/Governor.js

```diff
@@ -0,0 +1,444 @@
+    })
+
+    beforeEach(async () => {
+        await fixture.setUp()
+        ecTools = await ECTools.new()
+        await Governor.link("ECTools", ecTools.address)
+        governor = await Governor.new()
+        setUint256 = await SetUint256.new()
+    })
+
+    afterEach(async () => {
+        await fixture.tearDown()
+    })
+
+
+    const setUint256Tx = async (i, nonce, sender) => {
+        const txData = utils.arrayify(setUint256.contract.methods.setUint256(utils.bigNumberify(i)).encodeABI())
+        const dataHash = utils.solidityKeccak256(["address", "uint256", "uint256", "bytes"], [setUint256.address, utils.bigNumberify("0"), utils.bigNumberify(nonce), txData])
+        const sig = await web3.eth.sign(dataHash, sender)
+        return {data: utils.hexlify(txData), hash: dataHash, sig: sig}
+    }
+
+    describe("constructor", () => {
+        it("initializes state: owner", async () => {
+            assert.equal(await governor.owner(), accounts[0])
+        })
+    })
+
+    describe("stageUpdate", () => {
+        it("reverts when sender is not owner", async () => {
+            const {data} = await setUint256Tx("0", "1", accounts[0])
+
+            await truffleAssert.reverts(
+                governor.stage(
+                    {
+                        target: [setUint256.address],
+                        value: ["0"],
+                        data: [data],
+                        nonce: "1"
+                    },
+                    "0",
+                    {from: accounts[1]}
+                ),
+                "unauthorized",
+            )
+
+            assert.equal((await governor.nonce()).toString(), "0")
+        })
+
+        it("reverts when nonce is too low", async () => {
+            const {data} = await setUint256Tx("1", "0", accounts[0])
+
+            await truffleAssert.reverts(
+                governor.stage(
+                    {
+                        target: [setUint256.address],
+                        value: ["0"],
+                        data: [data],
+                        nonce: "0"
+                    },
+                    "0"
+                ),
+                "nonce too low"
+            )
+
+            assert.equal((await governor.nonce()).toString(), "0")
+        })
+
+        it("reverts when the current block number added by the delay overflows", async () => {
+            const {data} = await setUint256Tx("1", "1", accounts[0])
+
+            await truffleAssert.reverts(
+                governor.stage(
+                    {
+                        target: [setUint256.address],
+                        value: ["0"],
+                        data: [data],
+                        nonce: "1"
+                    },
+                    "115792089237316195423570985008687907853269984665640564039457584007913129639935"
+                )
+            )
+        })
+
+        it("stages an update: emit UpdateStaged event and increase updateCount and nonce", async () => {
```

Comment:
"updateCount" seems to imply there is an additional var that is being updated. If that is not the case I suggest removing it.

Reply:
fixed in 1de239d